### PR TITLE
Sample read blob events in the stress tests

### PR DESCRIPTION
**Pull request by andre4i**
## Description

ADO:5682

Read blob events account for 6% of all event types in the stress tests. We need to decrease the volume.

### CODE REVIEW

File: packages/utils/telemetry-utils/src/logger.ts [hunk 1]
```diff
@@ -708,6 +728,14 @@ export class PerformanceEvent {
...
 		this.logger.sendPerformanceEvent(event, error);
 	}
+
+	private static readonly eventHits = new Map<string, number>();
+	private static shouldReport(event: ITelemetryGenericEvent, sampleThreshold: number): boolean {
+		const eventKey = `.${event.category}.${event.eventName}`;
+		const hitCount = PerformanceEvent.eventHits.get(eventKey) ?? 0;
+		PerformanceEvent.eventHits.set(eventKey, hitCount + 1);
+		return hitCount % sampleThreshold === 0;
+	}
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/logger.ts:738]: Should we reset the hitCount to 0 when we hit the threshold, so a very high-volume event doesn't potentially end up overflowing it? Also, nice clean way to ensure the first event gets emitted, I missed it does it until I looked at the tests 👍.

File: packages/utils/telemetry-utils/src/logger.ts [hunk 2]
```diff
@@ -571,17 +571,25 @@ export class PerformanceEvent {
 		event: ITelemetryGenericEvent,
 		markers?: IPerformanceEventMarkers,
 		recordHeapSize: boolean = false,
+		emitLogs: boolean = true,
```
Comment (alexvy86) [inline packages/utils/telemetry-utils/src/logger.ts:574]: Let's add a bit of docs to the method to clarify that even if `emitLogs = false` the class will still produce performance markers. I was going to suggest removing the param

**Review by alexvy86 — APPROVED** [review]
Few small things but looks good otherwise.

File: packages/utils/telemetry-utils/src/logger.ts (hunk 1 above)
Comment (andre4i) [reply packages/utils/telemetry-utils/src/logger.ts:738]: good idea

File: packages/utils/telemetry-utils/src/logger.ts (hunk 2 above)
Comment (andre4i) [reply packages/utils/telemetry-utils/src/logger.ts:574]: I will delay this until after I test it in the stress tests as I need to figure out a good narrative about what the other parameters do 😄. I can't just add documentation only for this one.

**Comment by andre4i** [comment]
I will merge this only after testing it in a private branch in the stress tests, with some metrics proving the decrease in log volume.

**Comment by andre4i** [comment]
Testing this in a private branch, it's looking pretty good:

```
| where Data_eventName == "fluid:telemetry:RouterliciousDriver:readBlob_end"
| summarize count() by runId
```

runId | count_
-- | --
194834-tinylicious-local-ci-refs/heads/next | 33177
**194832-tinylicious-local-ci-refs/heads/test/andreiiacob** | **464**
194785-tinylicious-local-ci-refs/heads/release/v2int/6.4 | 49344
194762-tinylicious-local-ci-refs/heads/main | 28343
194729-tinylicious-local-ci-refs/heads/main | 33571
194715-tinylicious-local-ci-refs/heads/release/v2int/7.0 | 39044

I think that's definitely going to save some storage space!


@alexvy86 can you take another look? I did some changes to the stress test feature gate logic since you approved it.

**Review by NicholasCouri — APPROVED** [review]

**Review by alexvy86 — APPROVED** [review]
Disclaimer: I'm not super familiar with the stress test infra but this looks ok to me. And the bits in the driver itself are good.

**Pull request merged**